Applications/Spire/Source/KeyBindings/TaskKeysPage.cpp

```diff
@@ -0,0 +1,90 @@
+#include "Spire/KeyBindings/TaskKeysPage.hpp"
+#include "Spire/KeyBindings/TaskKeysTableView.hpp"
+#include "Spire/Ui/Button.hpp"
+#include "Spire/Ui/Icon.hpp"
+#include "Spire/Ui/SearchBox.hpp"
+#include "Spire/Ui/TextAreaBox.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto make_help_text_box() {
+    auto help_text_box = make_text_area_label(
+      QObject::tr("Allowed keys are: <b>F1–F12</b> and <b>Ctrl, Shift, "
+        "Alt  +  F1–F12</b> and <b>Ctrl, Shift, Alt  +  0–9</b>"));
+    help_text_box->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
+    update_style(*help_text_box, [] (auto& style) {
+      auto font = QFont("Roboto");
+      font.setWeight(QFont::Normal);
+      font.setPixelSize(scale_width(10));
+      style.get(Any()).
+        set(text_style(font, QColor(0x808080))).
+        set(horizontal_padding(scale_width(8))).
+        set(PaddingTop(scale_height(10))).
+        set(PaddingBottom(scale_height(5)));
+    });
+    return help_text_box;
+  }
+
+  auto make_button(const QString& path, const QString& tooltip) {
+    auto button = make_icon_button(imageFromSvg(path, scale(16, 16)), tooltip);
+    button->setFixedSize(scale(26, 26));
+    return button;
+  }
+}
+
+TaskKeysPage::TaskKeysPage(std::shared_ptr<KeyBindingsModel> key_bindings,
```

Comment:
Why is this a class? I feel like a repeating issue is the use of classes when they are not needed.

A class is used to encapsulate and manage both state and behavior. If all you need is state you use a struct, if all you need is behavior you use a function. If what you need is to manage both state and behavior then you use a class.

What state and behavior is needed for this?

Reply:
Since there is no state that the `TaskKeysPage` needs to manage, it is suitable to use a function to create it. However, the `TaskKeysPage` requires key press events to add and duplicate rows.  I found `KeyObserver` can monitor the key press events, but I don’t know how to use it in a function to monitor and get key events.